### DISCUSSION TITLE
Handling failures with selectFarthestJoined using Objects instead of Strings

### DISCUSSION THREAD

**Discussion by GP4cK** [petitparser/dart-petitparser#159] [Q&A] [answered]
Hello! First of all, thanks for this library. It's really great!

As I'm using `selectFarthestJoined`, when validating input, I sometimes get combined failures like `Expected date OR Expected duration`.
In this case, I'd like to display to the user `Expected date or duration`.
I could write some code that splits the failure around ` OR ` and then try to combine the failures together but I'm wondering if there's a better way... I could also JSON stringify the failure messages in the parsers so it's easier to combine later on.
Thanks for the help!

**Comment by renggli** [comment, marked as answer]
Since error messages are `String` values, there is (at least for now) no elegant way than to keep track of the combined error messages. Instead of using `selectFarthestJoined` you could of course [copy that small snippet of code](https://github.com/petitparser/dart-petitparser/blob/d395fd05e03b9ebd850dc718536b2d197fbb869d/lib/src/parser/utils/failure_joiner.dart#L19-L24) and create whatever combined error message you prefer?
